.github/release_pull_request_template.md

```diff
@@ -0,0 +1,42 @@
+This is the release pull request for {version} of the WooCommerce Blocks plugin.
+
+## Communication
+
+<!--
+  This section is for any notes related to communicating the release.
+  Please include any extra details with each item as needed.
+-->
+
+* [ ] What is being introduced in this release?
+* [ ]  If this release has potentially **high customer impact**, is a blog post (section) ready? Provide a link if applicable.
+* [ ] Is the **relevant developer documentation updated**? Please provide links if applicable.
+* [ ] Any special instructions or helpful notes for Happiness engineers are posted (this might not be needed for every release).
+
+## Quality
+
+<!--
+  This section is for any notes related to quality around the release.
+  Please include any extra details with each item as needed.
+-->
+
+* [ ] Have you **written tests** to cover the changes?
+     * [ ] Unit tests
+     * [ ] E2E tests
+     * [ ] for each supported WordPress and WooCommerce core versions.
+* [ ] Have you tested on the applicable platforms?
+     * [ ] mobile
+     * [ ] desktop
+* [ ] Does this release affect API's and conform to API versioning policy?
+* [ ] Have you considered the impact of the changes in the release on **other extensions** and **backward compatibility**?
+     * [ ] Does the release change the signature of any public methods or functions?
+     * [ ] Are filters or hooks affected by the PR code changes?
+* [ ] Does the release include a changelog entry in the readme.txt?
+* [ ] Please provide a link to **testing instructions** for this release.
```

Comment:
O, are these merged before the PR template is made? Or where do they live? As a file or wiki or both or where? If we're testing internally only, is the repo the best place for them verses p2 or master thread? Just curious.

Reply:
Good questions (and deserving some more explanation in the doc as well).

While there will be testing that happens during development, I view this section as any extra verification testing steps for things included in the release that is done before releasing. It will be up to the release lead reviewing the changeset to decide what these testing notes include and the confidence level they have in what needs manually tested.

I think I can do some improvement on wording around this both to provide a space for where the link to testing notes are (or instructions on where they go) and the checklist just indicates they've been added (if needed).